### ISSUE TITLE
‘freebsd-update ... fetch’ invocations should work when iocage is run non-interactively

### ISSUE THREAD

**Issue by dneades** [iocage/iocage#174]
I have just begun experimenting with iocage. I am trying to run `iocage fetch release=10.2-RELEASE` on a remote FreeBSD 10.2 machine over `ssh` using `salt-ssh` (http://www.saltstack.com/). This mostly works, except that the `freebsd-update ... fetch` fails, since `freebsd-update` detects it is running non-interactively and complains that `freebsd-update cron` should be used instead:

```
  INFO: Creating zroot/iocage
  INFO: Creating zroot/iocage/jails
  INFO: Creating zroot/iocage/.defaults
  INFO: Creating zroot/iocage/download
  INFO: Creating zroot/iocage/releases
Extracting: base.txz
Extracting: doc.txz
Extracting: lib32.txz
Extracting: src.txz
* Updating base jail..
freebsd-update fetch should not be run non-interactively.
Run freebsd-update cron instead.
No updates are available to install.
Run '/usr/sbin/freebsd-update fetch' first.

Creating basejail ZFS datasets... please wait.
```

Adding the `--not-running-from-cron` flag to the `freebsd-update ... fetch` command in `/usr/local/lib/iocage/ioc-common` fixes the problem:

```
        echo "* Updating base jail.."
        sleep 2

        env UNAME_r="$release" env PAGER="/bin/cat" \
            /usr/sbin/freebsd-update \
            -b $iocroot/releases/$release/root \
            -d $iocroot/releases/$release/root/var/db/freebsd-update/ --not-running-from-cron fetch
```

There’s a similar line in `/usr/local/lib/iocage/ioc-upgrade` that will no doubt also need tweaking:

```
        echo "* Updating jail.."
        env UNAME_r="$release" env PAGER="/bin/cat" \
            /usr/sbin/freebsd-update\
            -b ${mountpoint}/root \
            -d ${mountpoint}/root/var/db/freebsd-update/ --not-running-from-cron fetch
```

Note that, since `--not-running-from-cron` is a relatively new addition to the `freebsd-update` command, it may be necessary to avoid adding this flag when running on older versions of FreeBSD.

**Comment by pannon** [comment]
@dneades thanks for clarifying this - would you mind submitting a PR for this?

Thanks!

**Comment by dneades** [comment]
@pannon Try this: https://github.com/iocage/iocage/pull/175

(This is my first GitHub pull request, so I hope everything is in order!)

**Comment by skarekrow** [comment]
Closed via #175.
